apps/studio/data/storage/bucket-object-delete-mutation.ts

```diff
@@ -0,0 +1,37 @@
+import { useMutation } from '@tanstack/react-query'
+
+import { del, handleError } from 'data/fetchers'
+
+type DownloadBucketObjectParams = {
+  projectRef: string
+  bucketId?: string
+  paths: string[]
+}
+export const deleteBucketObject = async (
+  { projectRef, bucketId, paths }: DownloadBucketObjectParams,
+  signal?: AbortSignal
+) => {
+  if (!bucketId) throw new Error('bucketId is required')
+
+  const { data, error } = await del('/platform/storage/{ref}/buckets/{id}/objects', {
+    params: {
+      path: {
+        ref: projectRef,
+        id: bucketId,
+      },
+    },
+    body: {
+      paths,
+    },
+    signal,
+  })
+
+  if (error) handleError(error)
+  return data
+}
+
+export function useBucketObjectDeleteMutation() {
+  return useMutation({
+    mutationFn: deleteBucketObject,
+  })
+}
```

Comment:
For all these new mutation functions is there a reason why we're not passing through the options like in the other mutations? Makes it easy to pass `onSuccess()` from outside for example

Reply:
pushed up a change to make them consistent :) although they arent being used at the moment